python/lsst/cp/pipe/ptc.py

```diff
@@ -432,85 +485,141 @@ def funcAstier(self, pars, x):
+        initial clipping of data points that are more than
+        config.initialNonLinearityExclusionThreshold away from lying on a
+        straight line. This other step is necessary because the photon transfer
+        curve turns over catastrophically at very high flux (because saturation
+        drops the variance to ~0) and these far outliers cause the initial fit
+        to fail, meaning the sigma cannot be calculated to perform the
+        sigma-clipping.
 
         Parameters
         ----------
-        fitVectorsDicti : `dict`
-            Dictionary with exposure time, mean, and variance vectors in a tuple
+        dataset : `lsst.cp.pipe.ptc.PhotonTransferCurveDataset`
+            The dataset containing the means, variances and exposure times
         ptcFitType : `str`
-            Fit a 'POLYNOMIAL' (degree: 'polynomialFitDegree') or '
-            ASTIERAPPROXIMATION' to the PTC
-
-        Returns
-        -------
-        fitPtcDict : `dict`
-            Dictionary of the form fitPtcDict[amp] =
-            (meanVec, varVec, parsFit, parsFitErr, index)
-        nlDict : `dict`
-            Dictionary of the form nlDict[amp] =
-            (timeVec, meanVec, linResidual, parsFit, parsFitErr)
+            Fit a 'POLYNOMIAL' (degree: 'polynomialFitDegree') or
+            'ASTIERAPPROXIMATION' to the PTC
         """
-        if ptcFitType == 'ASTIERAPPROXIMATION':
-            ptcFunc = self.funcAstier
-            parsIniPtc = [-1e-9, 1.0, 10.]  # a00, gain, noise
-        if ptcFitType == 'POLYNOMIAL':
-            ptcFunc = self.funcPolynomial
-            parsIniPtc = np.repeat(1., self.config.polynomialFitDegree + 1)
-
-        parsIniNl = [1., 1., 1.]
-        fitPtcDict = {key: {} for key in fitVectorsDict}
-        nlDict = {key: {} for key in fitVectorsDict}
-        gainDict = {key: {} for key in fitVectorsDict}
-        noiseDict = {key: {} for key in fitVectorsDict}
 
         def errFunc(p, x, y):
             return ptcFunc(p, x) - y
 
+        sigmaCutPtcOutliers = self.config.sigmaCutPtcOutliers
         maxIterationsPtcOutliers = self.config.maxIterationsPtcOutliers
-        for amp in fitVectorsDict:
-            timeVec, meanVec, varVec = fitVectorsDict[amp]
-            timeVecOriginal = np.array(timeVec)
-            meanVecOriginal = np.array(meanVec)
-            varVecOriginal = np.array(varVec)
-            index0 = ((meanVecOriginal > self.config.minMeanSignal) &
-                      (meanVecOriginal <= self.config.maxMeanSignal))
-            #  Before bootstrap fit, do an iterative fit to get rid of outliers in PTC
+
+        for ampName in dataset.ampNames:
+            timeVecOriginal = np.array(dataset.rawExpTimes[ampName])
+            meanVecOriginal = np.array(dataset.rawMeans[ampName])
+            varVecOriginal = np.array(dataset.rawVars[ampName])
+
+            mask = ((meanVecOriginal > self.config.minMeanSignal) &
```

Comment:
Should it be ">=" self.config.minMeanSignal too? In either case, we should update the docstring for these two parameters ("self.config.minMeanSignal " and "self.config.maxMeanSignal") to say explicitly if the value of the parameter is included in the cut or not (sorry I did not do this originally).

Reply:
I don't know, I didn't change the logic in this block from your original code. The only reason this line appears in this PR is because the name of the variable it is assigned to changed from `index0` to `mask`.

Reply:
I've changed the logic to be inclusive for both bounds now, and updated the docstrings to  say so though.